From 49878e0759161d357782802f954886bae7bcc83b Mon Sep 17 00:00:00 2001
From: Diana Craciun <diana.craciun@nxp.com>
Date: Thu, 21 Nov 2019 15:06:47 +0200
Subject: [PATCH] bus/fsl-mc: Extend functions from the bus driver to be used
 by vfio-mc

The bus/dpcr driver use some common functions, export those
functions to be accessible from the vfio-mc driver.

Signed-off-by: Diana Craciun <diana.craciun@nxp.com>
---
 drivers/bus/fsl-mc/fsl-mc-private.h | 8 --------
 include/linux/fsl/mc.h              | 8 ++++++++
 2 files changed, 8 insertions(+), 8 deletions(-)

--- a/drivers/bus/fsl-mc/fsl-mc-private.h
+++ b/drivers/bus/fsl-mc/fsl-mc-private.h
@@ -182,14 +182,6 @@ int fsl_mc_msi_domain_alloc_irqs(struct
 
 void fsl_mc_msi_domain_free_irqs(struct device *dev);
 
-int __must_check fsl_create_mc_io(struct device *dev,
-				  phys_addr_t mc_portal_phys_addr,
-				  u32 mc_portal_size,
-				  struct fsl_mc_device *dpmcp_dev,
-				  u32 flags, struct fsl_mc_io **new_mc_io);
-
-void fsl_destroy_mc_io(struct fsl_mc_io *mc_io);
-
 bool fsl_mc_is_root_dprc(struct device *dev);
 
 #ifdef CONFIG_FSL_MC_UAPI_SUPPORT
--- a/include/linux/fsl/mc.h
+++ b/include/linux/fsl/mc.h
@@ -1013,6 +1013,14 @@ int dprc_scan_objects(struct fsl_mc_devi
 			  bool alloc_interrupts,
 		      unsigned int *total_irq_count);
 
+int __must_check fsl_create_mc_io(struct device *dev,
+				  phys_addr_t mc_portal_phys_addr,
+				  u32 mc_portal_size,
+				  struct fsl_mc_device *dpmcp_dev,
+				  u32 flags, struct fsl_mc_io **new_mc_io);
+
+void fsl_destroy_mc_io(struct fsl_mc_io *mc_io);
+
 int fsl_mc_find_msi_domain(struct device *mc_platform_dev,
 			   struct irq_domain **mc_msi_domain);
 
